hbase-common/src/main/java/org/apache/hadoop/hbase/HConstants.java

```diff
@@ -1040,10 +1040,21 @@ public enum OperationStatusCode {
   public static final String HFILE_PREAD_ALL_BYTES_ENABLED_KEY = "hfile.pread.all.bytes.enabled";
   public static final boolean HFILE_PREAD_ALL_BYTES_ENABLED_DEFAULT = false;
 
-  /*
-   * Minimum percentage of free heap necessary for a successful cluster startup.
+  /**
+   * Configuration key for the minimum fraction of heap memory that must remain free for a
+   * RegionServer to start
+   */
+  public static final String HBASE_REGION_SERVER_FREE_HEAP_MIN_SIZE_KEY =
+    "hbase.regionserver.free.heap.min.size";
+
+  public static final float HBASE_REGION_SERVER_FREE_HEAP_MIN_SIZE_DEFAULT = 0.1f;
```

Comment:
I'd recommend keeping the default unchanged unless there's a strong reason. Some servers may still have relatively limited memory (e.g. VM), so it's probably safer to keep the original value.